lib/Conversion/StructuredToMemref/StructuredToMemrefPass.cpp

```diff
@@ -64,17 +82,93 @@ class LoopTypeConverter : public TypeConverter {
     // The order of type conversion is important: later ones are tried earlier.
     addConversion([](Type type) { return type; });
     addConversion([context](triton::PointerType ptrType) {
-      SmallVector<int64_t> strides{1};
-      auto layout =
-          StridedLayoutAttr::get(context, ShapedType::kDynamic, strides);
-
-      auto elemType = ptrType.getPointeeType();
-      auto memrefType = MemRefType::get({1}, elemType, layout);
-      return memrefType;
+      return getMemrefTypeForScalarPtr(ptrType, context);
     });
   }
 };
 
+struct ScalarAddptrConverter
```

Comment:
i haven't figured out a good place to put this converter because it now runs before the other converters; so putting it in `StructuredToMemref.cpp` doesn't make sense either. I removed old version of converter in `StructuredToMemref.cpp`. Keeping this converter here for now, I'll update its location later.